src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs

```diff
@@ -93,76 +94,100 @@
-                                                                       (update :labels utils/delete i))))}
-                                    (icons/render-icon {} :remove))
-                                  [input/TextField {:style {:ref (str "key" i)
-                                                            :marginBottom 0 :width "47.5%" :marginRight "4%"}
-                                                    :defaultValue (first label)
-                                                    :onChange #(swap! state update-in [:labels i]
-                                                                      assoc 0 (-> % .-target .-value))}]
-                                  [input/TextField {:style {:ref (str "val" i)
-                                                            :marginBottom 0 :width "47.5%"}
-                                                    :defaultValue (last label)
-                                                    :onChange #(swap! state update-in [:labels i]
-                                                                      assoc 1 (-> % .-target .-value))}]
-                                  (common/clear-both)])
+                                 (flex/box {:style {:marginBottom 10}}
+                                   (links/create-internal
+                                     {:style {:color (:text-light style/colors)
+                                              :marginLeft -20
+                                              :minHeight 20 :minWidth 20
+                                              }
+                                      :href "javascript:;"
+                                      :onClick (fn [] (swap! state #(-> % (assoc :label-gensym (gensym))
+                                                                        (update :labels utils/delete i))))}
+                                     (icons/render-icon {:style {:marginTop "35%"}} :remove))
+                                   [input/TextField {:data-test-id (str "key-" i "-input")
+                                                     :style {:ref (str "key" i) :marginBottom 0 :width "48%" :marginRight "4%"}
+                                                     :defaultValue (first label)
+                                                     :onChange #(swap! state update-in [:labels i]
+                                                                       assoc 0 (-> % .-target .-value))}]
+                                   [input/TextField {:data-test-id (str "value-" i "-input")
+                                                     :style {:ref (str "val" i) :marginBottom 0 :width "48%"}
+                                                     :defaultValue (last label)
+                                                     :onChange #(swap! state update-in [:labels i]
+                                                                       assoc 1 (-> % .-target .-value))}]
+                                   (common/clear-both)))
```

Comment:
I don't see anything floating... is this `clear-both` necessary?

Reply:
You're right, I think that was a copy-pasta leftover :)